### PR TITLE
Create HTML Block Type with Coder Role

### PR DESCRIPTION
**Pull request by macburgee1**
- Create a new block type called "HTML Code" (or something else we can agree upon)
- Create a new text format called "HTML"
- Create a new role called "Coder"

Considerations:

- Show we allow <iframe>?
- Should we allow <script>?
- Should we allow <style>?
- If we allow <style>, should we provide a separate field that ensure the CSS is scoped to this particular block?

Blocked by:
#45 - need to enable Custom Blocks first
#46 - need block type permissions (including within Layout Builder)

### CODE REVIEW

**Comment by ericras** [comment]
Going fullscreen hides the codemirror editor behind the toolbar. 

fullscreen

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -118,6 +120,7 @@ theme:
   classy: 0
   seven: 0
   unl_five_herbie: 0
+  bartik: 0
```
Comment (ericras) [inline config/sync/core.extension.yml:123]: Bartik probably doesn't need enabling?

**Comment by ericras** [comment]
Regarding the full screen issue screenshot above:

Layout Builder Modal uses core modal functionality that utilizes JQuery UI's dialog.  It looks at everything that has class "ui-front" and makes the modal's z-index one greater.  However, the toolbar doesn't have "ui-front" (and my attempt at patching it to add "ui-front" didn't end up working anyway).

(I was thinking that this will impact things in a future 9.x release but on second thought I'm not sure:  https://www.drupal.org/project/drupal/issues/2158943)

It looks like the "solution" in Seven & Bartik is simply `.ui-dialog {z-index:1260}` in their theme's dialog.css

File: config/sync/system.performance.yml [hunk 2]
```diff
@@ -2,15 +2,15 @@ cache:
   page:
     max_age: 0
 css:
-  preprocess: true
+  preprocess: false
```
Comment (ericras) [inline config/sync/system.performance.yml:5]: Looks like these two settings are going to get exported by mistake out of a dev environment every time. Do these preprocess settings for css/js need to be included in config_ignore now? Or - does settings.php.inc in the herbie profile need them moved from the development section to the production section?

File: web/modules/custom/unl_blocks/unl_blocks.module [hunk 3]
```diff
@@ -6,6 +6,26 @@
  */
 
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\unl_utility\UNLUtilityTrait;
+
+/**
+ * Implements hook_form_alter().
+ */
+function unl_blocks_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Drupal does not respect JS dependencies when loading loading ia AJAX.
```
Comment (ericras) [inline web/modules/custom/unl_blocks/unl_blocks.module:15]: typo

**Comment by ericras** [comment]
CSS does not insert nor update when in layout builder's preview mode. A full save action is needed to see the results of the CSS.  I don't think that's a blocker to move forward here.  Issue on drupal.org: https://www.drupal.org/project/field_css/issues/3154655

File: web/modules/custom/unl_blocks/unl_blocks.module (hunk 3 above)
Comment (macburgee1) [reply web/modules/custom/unl_blocks/unl_blocks.module]: Typo is fixed and pushed

File: config/sync/system.performance.yml (hunk 2 above)
Comment (macburgee1) [reply config/sync/system.performance.yml:5]: I think this is handled in master but hasn't be been merged into this branch yet. I've fixed the errant config change in this branch

File: config/sync/core.extension.yml (hunk 1 above)
Comment (macburgee1) [reply config/sync/core.extension.yml]: That was an errant config export. It's been fixed and pushed.

File: web/modules/custom/unl_blocks/unl_blocks.module [hunk 4]
```diff
@@ -12,7 +12,7 @@ use Drupal\unl_utility\UNLUtilityTrait;
  * Implements hook_form_alter().
  */
 function unl_blocks_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  // Drupal does not respect JS dependencies when loading loading ia AJAX.
+  // Drupal does not respect JS dependencies when loading loading via AJAX.
```
Comment (ericras) [inline web/modules/custom/unl_blocks/unl_blocks.module:15]: typo still: "loading loading"

Comment (macburgee1) [reply web/modules/custom/unl_blocks/unl_blocks.module]: Oops. Fixed and pushed.

**Comment by macburgee1** [comment]
I'm going to lookin into the Z-index issue and the Field CSS issues

**Comment by macburgee1** [comment]
The Field CSS module has been updated and now supports Layout Builder's live updates. Changes made to CSS field on the HTML Code block will now update immediately in life preview.

**Pull request merged**